File: contracts/dividend/schema.rs
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

use wasplib::client::*;

const SC_NAME: &str = "dividend";
const SC_HNAME: Hname = Hname(0xcce2e239);

const PARAM_ADDRESS: &str = "address";
const PARAM_FACTOR: &str = "factor";

const VAR_MEMBERS: &str = "members";
const VAR_TOTAL_FACTOR: &str = "total_factor";

const FUNC_DIVIDE: &str = "divide";
const FUNC_MEMBER: &str = "member";

const HFUNC_DIVIDE: Hname = Hname(0xc7878107);
const HFUNC_MEMBER: Hname = Hname(0xc07da2cb);

File: contracts/dividend/schema.go
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

package dividend

import "github.com/iotaledger/wasplib/client"

const ScName = "dividend"
const ScHname = client.Hname(0xcce2e239)

const ParamAddress = client.Key("address")
const ParamFactor = client.Key("factor")

const VarMembers = client.Key("members")
const VarTotalFactor = client.Key("total_factor")

const FuncDivide = "divide"
const FuncMember = "member"

const HFuncDivide = client.Hname(0xc7878107)
const HFuncMember = client.Hname(0xc07da2cb)

func OnLoad() {
	exports := client.NewScExports()
	exports.AddCall(FuncDivide, funcDivide)
	exports.AddCall(FuncMember, funcMember)
}
